server/remote_cache/byte_stream_server/byte_stream_server.go

```diff
@@ -333,16 +338,26 @@ func (s *ByteStreamServer) beginWrite(ctx context.Context, req *bspb.WriteReques
 				return nil, err
 			}
 			ws.writer = io.MultiWriter(decompressingChecksum, committedWriteCloser)
-			ws.decompressorCloser = decompressingChecksum
+			ws.bufioCloser = decompressingChecksum
 		} else {
 			// If the cache doesn't support compression, wrap both the checksum and cache writer in a decompressor
 			decompressor, err := compression.NewZstdDecompressor(ws.writer)
 			if err != nil {
 				return nil, err
 			}
 			ws.writer = decompressor
-			ws.decompressorCloser = decompressor
+			ws.bufioCloser = decompressor
 		}
+	} else if compressData {
```

Comment:
nit: On L334 can you add to the comment that the data is already compressed? Took me a minute to figure out why these cases are different